addons/addon-base-raas/packages/base-raas-services/lib/migration/migration-service.js

```diff
@@ -0,0 +1,113 @@
+ *  permissions and limitations under the License.
+ */
+
+const Service = require('@aws-ee/base-services-container/lib/service');
+const { isAdmin } = require('@aws-ee/base-services/lib/authorization/authorization-utils');
+
+const settingKeys = { studiesTableName: 'dbStudies', categoryIndexName: 'dbStudiesCategoryIndex' };
+
+class MigrationService extends Service {
+  constructor() {
+    super();
+    this.dependency(['studyOperationService', 'dbService', 'userService', 'studyService']);
+  }
+
+  async init() {
+    await super.init();
+    const dbService = await this.service('dbService');
+
+    const studiesTable = this.settings.get(settingKeys.studiesTableName);
+    this._getter = () => dbService.helper.getter().table(studiesTable);
+    this._updater = () => dbService.helper.updater().table(studiesTable);
+    this._query = () => dbService.helper.query().table(studiesTable);
+    this._deleter = () => dbService.helper.deleter().table(studiesTable);
+    this._scanner = () => dbService.helper.scanner().table(studiesTable);
+
+    this.categoryIndex = this.settings.get(settingKeys.categoryIndexName);
+  }
+
+  async migrateMyStudiesPermissions(requestContext, migrationMappings) {
+    if (!isAdmin(requestContext)) {
+      throw this.boom.forbidden('You need admin permissions to migrate My Studies', true);
+    }
+    const studyOperationService = await this.service('studyOperationService');
+    const userService = await this.service('userService');
+    const studyService = await this.service('studyService');
+
+    const migrationResults = [];
+    await Promise.all(
+      migrationMappings.map(async item => {
+        const endUser = item.uid;
+        const studyId = item.studyId;
+        if (await userService.isInternalAuthUser(endUser)) {
+          throw this.boom.badRequest('You cannot migrate My Studies to internal user', true);
+        }
+
+        // check that end user is not already owner
+        if (await studyService.isStudyAdmin(requestContext, studyId, endUser)) {
+          throw this.boom.badRequest(`Current study is already owned by ${endUser}`, true);
+        }
+
+        const result = await studyOperationService.updatePermissions(requestContext, studyId, {
+          usersToAdd: [{ uid: endUser, permissionLevel: 'admin' }],
+          usersToRemove: [{ uid: '*', permissionLevel: 'admin' }],
+        });
+        migrationResults.push(result);
+      }),
+    );
+    return migrationResults;
+  }
+
+  async listMyStudies(requestContext) {
+    if (!isAdmin(requestContext)) {
+      throw this.boom.forbidden('You need admin permissions to list all My Studies in this environment', true);
+    }
+
+    const studyService = await this.service('studyService');
+    const userService = await this.service('userService');
+
+    const result = await this._query()
+      .index(this.categoryIndex)
+      .key('category', 'My Studies')
+      .limit(1000)
```

Comment:
1. Will this API be hit from the UI? 
2. How likely it is to hit the limit?

Reply:
1. The API will not be hit from the UI.
2. This is the same limit on the method called from GET /studies (`list()` in `study-service.js`). If we are accepting this limit there, I thought it would be okay to accept the limit here.